### ISSUE TITLE
Add training smoke test: assert loss decreases after 5 steps on dummy data

### ISSUE THREAD

**Issue by joemunene-by** [joemunene-by/GhostLM#16]
## Problem

We have 16 unit tests covering model forward passes, attention mechanics, tokenizer behaviour, and weight tying — but nothing actually exercises the training loop end-to-end. If a PR breaks the backward pass, optimizer step, or LR schedule in a subtle way, CI passes anyway.

## Proposed change

Add `tests/test_training_smoke.py` with one test: train for 5 steps on tiny in-memory data and assert that loss at step 5 is lower than loss at step 0.

```python
def test_training_loss_decreases():
    cfg = GhostConfig(n_layers=2, d_model=64, n_heads=2, block_size=32, ...)
    model = GhostLM(cfg)
    
    # Tiny deterministic dataset — e.g., 10 random token sequences
    torch.manual_seed(42)
    data = torch.randint(0, cfg.vocab_size, (10, cfg.block_size))
    
    losses = []
    optimizer = model.configure_optimizers(weight_decay=0.01, lr=1e-3)
    for step in range(5):
        x, y = data[:, :-1], data[:, 1:]
        logits, loss = model(x, targets=y)
        losses.append(loss.item())
        optimizer.zero_grad(); loss.backward(); optimizer.step()
    
    assert losses[-1] < losses[0], f"Loss did not decrease: {losses}"
```

## Why this matters

This is the cheapest possible regression test for the entire training pipeline. Takes maybe 10 seconds to run, catches a huge class of bugs (broken gradients, wrong loss reduction, LR=0 bugs, frozen parameters).

## Pointers

- `ghostlm/trainer.py` has `configure_optimizers` — reuse it
- Keep `n_layers=2`, `d_model=64`, `block_size=32` so the test runs fast on CPU in CI

**Comment by joemunene-by** [comment]
Resolved in 9bdc813. Added `tests/test_training_smoke.py` with two end-to-end tests: `test_loss_decreases_on_dummy_data` (5 optimizer steps must drive loss down on synthetic data) and `test_grad_flows_through_all_blocks` (every grad-required parameter receives a gradient after one backward pass). Both pass in under 11s on CPU. The full suite is now 71 tests, all passing.
